src/Metrics/MetricsBucket.php

```diff
@@ -2,81 +2,7 @@
 
 namespace Unleash\Client\Metrics;
 
-use DateTimeInterface;
-use JetBrains\PhpStorm\ArrayShape;
-use JsonSerializable;
-use LogicException;
-
-/**
- * @internal
- */
-final class MetricsBucket implements JsonSerializable
+interface MetricsBucket
```

Comment:
I'd personally add the JsonSerializable interface here, or is there any reason why not to? Also this interface definitely needs some methods that will be used by the DefaultMetricsSender - it should work with any instance of `MetricsBucket`, meaning you need to declare all common methods here.

```suggestion
use JsonSerializable;

interface MetricsBucket extends JsonSerializable
```

Reply:
The `JsonSerializable` interface is only needed in case you want to customise the serialisation output, it is not a prerequisite to use `json_encode` on an object.

As for the functions, the `DefaultMetricsSender` only needs to serialise the bucket with `json_encode`. The `MetricsBucket` is a simple DTO, is there really any function needed to be exposed ?

Reply:
Every JSON serializable object in the SDK uses the interface directly, this should be no exception.

As for the methods, yes, they are needed if you want to work with them in a universal way. If the interface is going to be public, it should definitely have getters, setters and a method to add a toggle to the bucket. Also a method that returns the data.

Reply:
> Every JSON serializable object in the SDK uses the interface directly, this should be no exception.

Alright, I'll add for consistency then 👍